### ISSUE TITLE
LiveMetrics(QuickPulse) need to try to filter out dependencies to QuickPulse service

### ISSUE THREAD

**Issue by cijothomas** [Microsoft/ApplicationInsights-dotnet-server#1095]
QuickPulseTelemetryProcessor, which is added **before** sampling and all other telemetry processors by default, does examine every dependency item to check if it is a call made to quickpulse service itself.
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/PerformanceCollector/Perf.Shared/QuickPulseTelemetryProcessor.cs#L151

However, the DependencyCollection module already contains logic to exclude all calls to any AI backend.
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/DependencyCollector/Shared/Implementation/ApplicationInsightsUrlFilter.cs#L89

The additional check in QuickPulseTelemetryProcessor may have been added as a an extra safety check, or may be added when DependencyCollector had unresolved bugs or something.

Proposing to remove this additional check in QuickPulseTelemetryProcessor as it is causing perf issue for apps making a ton of outbound calls, as it does `IndexOf` operation on the target for **every item** , as its added before sampling takes place.

**Comment by cijothomas** [comment]
@tokaplan Any comments on this?

**Comment by tokaplan** [comment]
Two situations to keep in mind here
1. QuickPulse is inactive. In this case, if DependencyCollector telemetry processor has logic to filter out calls to any AI backend, including QPS, it is indeed a good idea to remove this check. This was never brought up at the time, so maybe DependencyCollector did not have that logic back then, or it simply didn't occur to anyone to add QPS to that list.
2. QuickPulse is active. In this case, QuickPulseTelemetryProcessor will not be affected by DependencyCollector in terms of which data it considers and which it doesn't, so the check must still be in place.

So to sum up - it should be totally fine to change the logic so that when QuickPulse is inactive it doesn't perform the check (if DependencyCollector indeed filters out calls to QPS). When QuickPulse is active - the check must still be performed.

Both scenarios must be tested :)

**Comment by cijothomas** [comment]
This was done in https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/401/files which was mid 2017. The original check was added prior to that.

I will do manual testing to ensure this is the case before removing the additional check anyway.
